### PR TITLE
Add boskos metrics to monitoring.prow.k8s.io

### PR DESCRIPTION
**Pull request by detiber**
Attempt to add a dashboard for boskos metrics to monitoring.prow.k8s.io

### CODE REVIEW

File: prow/cluster/monitoring/README.md [hunk 1]
```diff
@@ -68,11 +68,6 @@ Developing a new dashboard can be achieved by
 
 * Add `bazel` target to [dashboards_out/BUILD.bazel](grafana_dashboards/BUILD.bazel) for generating the configMap with the json file above.
 
-    ```
-    ### if you want to apply the configMaps
-    $ bazel run //prow/cluster/monitoring/mixins/dashboards_out:grafana-configmaps.apply
```
Comment (detiber) [inline prow/cluster/monitoring/README.md]: @fejta this bazel target no longer exists as of https://github.com/kubernetes/test-infra/commit/1448acef54ce9c2ab4b25dcd13bc66eb69a7af46#diff-5bee9d18e9a2ba62086a15a711aba3d1

It wasn't clear if there was an alternative target I should replace this with for the docs (and for testing).

**Comment by cblecker** [comment]
/cc @stevekuznetsov 
I'm not sure what the deployment procedure is for these dashboards.

**Comment by detiber** [comment]
> I'm not sure what the deployment procedure is for these dashboards.

It's supposed to be automated based on the discussion I had with @fejta: https://kubernetes.slack.com/archives/C09QZ4DQB/p1570728227013100

File: prow/cluster/monitoring/prow_servicemonitors.yaml [hunk 2]
```diff
@@ -169,3 +169,22 @@ spec:
...
+  name: boskos
+  namespace: prow-monitoring
+spec:
+  endpoints:
+    - interval: 30s
+      port: default
+      scheme: http
+  namespaceSelector:
+    matchNames:
+      - test-pods
```
Comment (stevekuznetsov) [inline prow/cluster/monitoring/prow_servicemonitors.yaml]: Isn't `boskos-metrics` running in the build cluster, not the service cluster?

Comment (detiber) [reply prow/cluster/monitoring/prow_servicemonitors.yaml]: Good question.

Assuming that is the case, is there currently a way to define talking to an external service/cluster using this new monitoring framework? The docs are currently limited to the use of a ServiceMonitor.

Comment (stevekuznetsov) [reply prow/cluster/monitoring/prow_servicemonitors.yaml]: You could have a `Service` in the service cluster that ran nginx or whatever to forward to the other location ...

Comment (stevekuznetsov) [reply prow/cluster/monitoring/prow_servicemonitors.yaml]: That's kind of a hack though -- I think preferably we would deploy prometheus on the build cluster and scrape there, then configure that as a backend for grafana to pull from

Comment (stevekuznetsov) [reply prow/cluster/monitoring/prow_servicemonitors.yaml]: We will likely need to use [auth](https://grafana.com/docs/plugins/developing/auth-for-datasources/) on that link

**Comment by fejta** [comment]
re deployment: I'm mostly parroting what I understand from @hongkailiu and @stevekuznetsov who have been doing the most work in the monitoring space (thanks :) )

File: prow/cluster/monitoring/prow_servicemonitors.yaml (hunk 2 above)
Comment (fejta) [reply prow/cluster/monitoring/prow_servicemonitors.yaml]: Yes, boskos is running in the test-pods namespace in different k8s cluster than the rest of prow...

Comment (hongkailiu) [reply prow/cluster/monitoring/prow_servicemonitors.yaml]: Prometheus itself does not [provide auth](https://prometheus.io/docs/guides/basic-auth/).
I feel an auth proxy like nginx is kind of unavoidable.
Prometheus and app staying in the same cluster sounds good to me and use grafana to join the dashboards.

**Comment by detiber** [comment]
Closing since this has been implemented in a separate PR.
/close

**Pull request closed without merging**